### ISSUE TITLE
Improved Leader of the pack/earthshield/Prayer of Mending Aggro Bug (BB #308)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#308]
This issue was migrated from bitbucket.
**Original Reporter:** 
**Original Date:** 30.09.2010 12:19:46 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/308

<hr>

The aggro generated by those Spells should count
for the target they were casted on, but instead 
the aggro goes right back to the original caster.

Very annoying for feraldruids who have massive aggro
issues because of that.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** 
**Original Date:** 04.10.2010 23:36:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/308/#comment-260472

<hr>

confirmed

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** 
**Original Date:** 25.12.2010 18:36:03 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/308/#comment-324815

<hr>

cmon.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 24.06.2013 02:53:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/308/#comment-4944530

<hr>

The below fix works. **_Changes made to Spell.cpp**_

```
#!diff

         int32 gain = unitTarget->ModifyHealth(int32(addhealth));

+       if(m_spellInfo->Id == 379 || m_spellInfo->Id == 33110)
+           unitTarget->getHostileRefManager().threatAssist(unitTarget, float(gain) * 0.5f, m_spellInfo);
+       else 
         unitTarget->getHostileRefManager().threatAssist(caster, float(gain) * 0.5f, m_spellInfo);
         if (caster->GetTypeId() == TYPEID_PLAYER)
             if (BattleGround *bg = caster->ToPlayer()->GetBattleGround())

```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 22.02.2014 18:44:27 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/308/#comment-8533711

<hr>

Fixed threat issue with heal-on-hit auras - resolves #308

→ <<cset dc963613c6af>>
